File: ekiv/fsharp3/Controllers/WeatherForecastController.fs
namespace fsharp3.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging
open fsharp3
open System.Net.Http

[<ApiController>]
[<Route("third")>]
type WeatherForecastController (logger : ILogger<WeatherForecastController>) =
    inherit ControllerBase()

    [<HttpGet>]
    member __.Get() : string =
        let client = new HttpClient()
        let res = client.GetStringAsync("http://localhost:8282/second").GetAwaiter().GetResult()
        if String.Equals(res, "Мы сломали", StringComparison.InvariantCultureIgnoreCase)
        then res + " твой"
        else ""

File: ekiv/csharp2/Controllers/WeatherForecastController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace csharp2.Controllers
{
    [ApiController]
    [Route("second")]
    public class WeatherForecastController : ControllerBase
    {
        [HttpGet]
        public async Task<string> Get()
        {
	try{
            var res = await new HttpClient().GetStringAsync("http://localhost:8080/");
            if (String.Equals(res, "Мы", StringComparison.InvariantCultureIgnoreCase))
            {
                return res + " сломали";
            }
            return "not equals";
}
catch(Exception e){
return "Exception" + e.ToString();
}
        }
    }
}
